modules/core/src/main/java/org/apache/ignite/internal/commandline/meta/subcommands/MetadataUpdateCommand.java

```diff
@@ -0,0 +1,125 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.commandline.meta.subcommands;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.logging.Logger;
+import org.apache.ignite.internal.binary.BinaryMetadata;
+import org.apache.ignite.internal.client.GridClient;
+import org.apache.ignite.internal.client.GridClientCompute;
+import org.apache.ignite.internal.client.GridClientConfiguration;
+import org.apache.ignite.internal.client.GridClientDisconnectedException;
+import org.apache.ignite.internal.client.GridClientNode;
+import org.apache.ignite.internal.commandline.CommandArgIterator;
+import org.apache.ignite.internal.commandline.meta.MetadataSubCommandsList;
+import org.apache.ignite.internal.commandline.meta.tasks.MetadataMarshalled;
+import org.apache.ignite.internal.commandline.meta.tasks.MetadataUpdateTask;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.internal.visor.VisorTaskArgument;
+
+/** */
+public class MetadataUpdateCommand
+    extends MetadataAbstractSubCommand<MetadataMarshalled, MetadataMarshalled>
+{
+    /** Output file name. */
+    public static final String IN_FILE_NAME = "--in";
+
+    /** {@inheritDoc} */
+    @Override protected String taskName() {
+        return MetadataUpdateTask.class.getName();
+    }
+
+    /** {@inheritDoc} */
+    @Override public String confirmationPrompt() {
+        return "Warning: the command will update the binary metadata at the cluster.";
+    }
+
+    /** {@inheritDoc} */
+    @Override public MetadataMarshalled parseArguments0(CommandArgIterator argIter) {
+        String opt = argIter.nextArg("--in");
+
+        if (!IN_FILE_NAME.equalsIgnoreCase(opt))
+            throw new IllegalArgumentException("");
+
+        Path inFile = FS.getPath(argIter.nextArg("input file name"));
+
+        try (InputStream is = Files.newInputStream(inFile)) {
+            ByteArrayOutputStream buf = new ByteArrayOutputStream();
+
+            U.copy(is, buf);
+
+            return new MetadataMarshalled(buf.toByteArray(), null);
+        }
+        catch (IOException e) {
+            throw new IllegalArgumentException("Cannot read metadata from " + inFile, e);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override protected MetadataMarshalled execute0(
```

Comment:
Look like all commands have the same execute0() methods, except logic for finding nodes from GridClientCompute, so let's change this part

Reply:
Great issue! Thanks a lot. Fixed.